pycontrails/core/flight.py

```diff
@@ -1112,6 +1113,101 @@ def filter_altitude(
         out.data.pop("level", None)  # avoid any ambiguity
         return out
 
+    def distance_to_coords(self: Flight, distance: ArrayOrFloat) -> tuple[
+        ArrayOrFloat,
+        ArrayOrFloat,
+        np.signedinteger[Any] | npt.NDArray[np.signedinteger[Any]],
```

Comment:
```suggestion
        np.intp | npt.NDArray[np.intp],
```

`intp` is a shorthand for segment indexes